core/src/databases/database.rs

```diff
@@ -38,16 +38,18 @@ pub enum QueryDatabaseError {
     GenericError(#[from] anyhow::Error),
     #[error("Too many result rows")]
     TooManyResultRows,
+    #[error("Query execution error: {0}")]
```

Comment:
Is this enough for the pattern matching?

Reply:
We actually just match against the `ExecutionError` variant which contains a string which is the SQLite error. This thing is just a way to very quickly impl the `Display` trait in case someone tries to print / stringify this error